consensus/wal.go

```diff
@@ -166,17 +167,25 @@ func (wal *baseWAL) SearchForEndHeight(height int64, options *WALSearchOptions)
 		for {
 			msg, err = dec.Decode()
 			if err == io.EOF {
+				// OPTIMISATION: no need to look for height in older files if we've seen height - 1
+				if lastHeightFound == height-1 {
```

Comment:
what if theres multiple EndHeight in a single file ? Shouldn't we just check `lastHeightFound > 0 && lastHeightFound < height` ?